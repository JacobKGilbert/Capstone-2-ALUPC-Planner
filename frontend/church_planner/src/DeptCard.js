import { Link } from 'react-router-dom'
import {
  Button,
  Card,
  CardBody,
  CardHeader,
  CardFooter,
  Col,
  Modal,
  ModalHeader,
  ModalBody,
  ModalFooter,
} from 'reactstrap'
import ChurchPlannerApi from './api'
import { useState } from 'react'

const DeptCard = ({ dept }) => {
  const [modal, setModal] = useState(false)
  const toggle = () => setModal(!modal)
  const handleClick = async () => {
    await ChurchPlannerApi.deleteDept(dept.code)
  }

  return (
    <Col sm={6}>
      <Card>
        <CardHeader tag="h3">
          <Link to={`${dept.code}`} className="text-decoration-none text-dark">
            {dept.name}
          </Link>
        </CardHeader>
        <CardBody tag="h6">
          Department Head: {dept.deptHead ? 
              <Link 
                to={`/users/${dept.deptHead.id}`}
              >
                {dept.deptHead.firstName} {dept.deptHead.lastName}
              </Link> : 
              'Not Set'
            }
        </CardBody>
        <CardFooter>
          <Button color="danger" className="pb-2" onClick={toggle}>
            Delete Department
          </Button>
        </CardFooter>

        <Modal isOpen={modal} toggle={toggle}>
          <ModalHeader toggle={toggle}>Are You Sure?</ModalHeader>
          <ModalBody>This action cannot be undone.</ModalBody>
          <ModalFooter>
            <Button color="danger" onClick={handleClick}>
              Delete Department
            </Button>{' '}
            <Button color="secondary" onClick={toggle}>
              Cancel
            </Button>
          </ModalFooter>
        </Modal>
      </Card>
    </Col>
  )
}

export default DeptCard
