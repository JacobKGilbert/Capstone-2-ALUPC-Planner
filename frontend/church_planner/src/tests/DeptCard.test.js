import React from 'react'
import { screen, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import DeptCard from '../DeptCard'
import useRouter from './renderWithRouter'

const testDept = {
  code: 'test',
  deptHead: {firstName: 'Test', lastName: 'Head'}
}

it('should render', () => {
  const { container } = render(useRouter(<DeptCard dept={testDept} />))

  expect(container).toHaveTextContent('Department Head: Test Head')
})

it('should match snapshot', () => {
  const { asFragment } = render(useRouter(<DeptCard dept={testDept} />))
  expect(asFragment()).toMatchSnapshot()
})

it('should display modal when delete button is clicked', () => {
  render(useRouter(<DeptCard dept={testDept} />))
  
  const modalToggleButton = screen.getByRole('button', {
    name: 'Delete Department',
  })

  userEvent.click(modalToggleButton)

  expect(screen.getByText('Are You Sure?')).toHaveTextContent('Are You Sure?')
})