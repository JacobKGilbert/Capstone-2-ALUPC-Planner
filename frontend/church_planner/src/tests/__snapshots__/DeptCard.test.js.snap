// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-sm-6"
  >
    <div
      class="card"
    >
      <h3
        class="card-header"
      >
        <a
          class="text-decoration-none text-dark"
          href="/test"
        />
      </h3>
      <h6
        class="card-body"
      >
        Department Head: 
        <a
          href="/users/undefined"
        >
          Test Head
        </a>
      </h6>
      <div
        class="card-footer"
      >
        <button
          class="pb-2 btn btn-danger"
          type="button"
        >
          Delete Department
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
